roles/setup_replication/tasks/setup_replication.yml

```diff
@@ -3,42 +3,95 @@
 
 - name: Set force_replication based on the user input
   set_fact:
-      force_replication: "{{ item.force_replication }}"
-  when: item.force_replication is defined
+      force_replication: "{{ hostvars[inventory_hostname].force_replication }}"
+  when: hostvars[inventory_hostname].force_replication is defined
   register: output
 
 - name: Force cleanup based on force_replication
   block:
     - include_tasks: rm_replication.yml
   when: force_replication
   become: yes
-  delegate_to: "{{ item.public_ip }}"
 
 # postgres
+- name: Gather the host and information for /etc/hosts
+  set_fact:
+    etc_hosts_lists: "{{ etc_hosts_lists + [node.private_ip + ' ' + node.hostname] }}"
+  loop: "{{ lookup('pg_sr_cluster_nodes', wantlist=True) }}"
+  loop_control:
+    loop_var: node
+  when: use_hostname
+  no_log: "{{ disable_logging }}"
+
+- name: Get the primary information
+  set_fact:
+    primary_public_ip: "{{ node.ansible_host }}"
+  loop: "{{ lookup('pg_sr_cluster_nodes', wantlist=True) }}"
+  loop_control:
+    loop_var: node
+  when:  node.node_type == 'primary'
+  no_log: "{{ disable_logging }}"
+
+- name: Prepare host ip lists based on allow_node_ips
+  include_tasks: prepare_hba_value_list.yml
+  loop: "{{ lookup('pg_sr_cluster_nodes', wantlist=True) }}"
+  loop_control:
+    loop_var: node
+  no_log: "{{ disable_logging }}"
+
+- name: Prepare synchronous_standby_names
+  set_fact:
+    synchronous_standbys: "{{ synchronous_standbys + [node.inventory_hostname | regex_replace('[^a-zA-Z0-9_]', '_')] }}"
+  loop: "{{ lookup('pg_sr_cluster_nodes', wantlist=True) }}"
+  loop_control:
+    loop_var: node
+  when:
+    - node.node_type == 'standby'
+    - node.replication_type == 'synchronous'
+  no_log: "{{ disable_logging }}"
+
+- name: Update primary for replication
+  import_tasks: primary_settings.yml
+  run_once: true
+  no_log: "{{ disable_logging }}"
+  delegate_to: "{{ primary_public_ip }}"
 
 - name: Update etc_hosts based on use_hostname
   block:
     - import_tasks: linux_update_etc_hosts.yml
   become: yes
+  no_log: "{{ disable_logging }}"
   when: use_hostname
-  delegate_to: "{{ item.public_ip }}"
 
 - name: Call upstream update based on the upstream node
   block:
     - import_tasks: upstream_node.yml
   when:
-    - item.upstream_node_private_ip is defined
+    - hostvars[inventory_hostname].upstream_node_private_ip is defined
+  no_log: "{{ disable_logging }}"
 
 - name: Build standby service check
   block:
     - import_tasks: create_directories.yml
     - import_tasks: pg_basebackup.yml
     - import_tasks: configure_node.yml
   become: yes
-  delegate_to: "{{ item.public_ip }}"
 
-- name: Reset the the variables based on the user input
+- name: Update primary for replication
+  import_tasks: primary_synchronous_param.yml
+  when: synchronous_standbys|length > 0
+  run_once: true
+  delegate_to: "{{ primary_public_ip }}"
```

Comment:
Same comment here, do we need to execute this for each standby ?

Reply:
This will be running only once for the cluster. I have included `run_once: true`

Reply:
Alright, nice.